src/assets/wise5/components/match/match-student/match-student-choice-reuse/match-student-choice-reuse.ts

```diff
@@ -15,4 +16,34 @@ export class MatchStudentChoiceReuse extends MatchStudentDefault {
     event.container.element.nativeElement.classList.remove('primary-bg');
     this.studentDataChanged();
   }
+
+  protected addAuthoredChoiceToBucket(choiceId: string, bucket: any): void {
+    bucket.items.push(copy(this.matchService.getChoiceById(choiceId, this.choices)));
+  }
+
+  protected getChoicesThatChangedSinceLastSubmit(latestSubmitComponentState: any): string[] {
+    const choicesThatChanged = super.getChoicesThatChangedSinceLastSubmit(
+      latestSubmitComponentState
+    );
+    const previousBuckets = latestSubmitComponentState.studentData.buckets;
+    for (const currentBucket of this.getNonSourceBuckets()) {
+      const {
+        currentBucketChoiceIds,
+        previousBucketChoiceIds
+      } = this.getPreviousAndCurrentChoiceIds(previousBuckets, currentBucket);
+      const choicesThatWereRemoved = previousBucketChoiceIds.filter(
+        (choiceId: string) => !currentBucketChoiceIds.includes(choiceId)
+      );
+      choicesThatChanged.push(...choicesThatWereRemoved);
+    }
+    return choicesThatChanged;
```

Comment:
Use declarative function (flatMap) to make this easier to read?
```suggestion
    const previousBuckets = latestSubmitComponentState.studentData.buckets;
    const removedChoices = this.getNonSourceBuckets().flatMap((bucket: any) => {
      const {
        currentBucketChoiceIds,
        previousBucketChoiceIds
      } = this.getPreviousAndCurrentChoiceIds(previousBuckets, bucket);
      return previousBucketChoiceIds.filter(
        (choiceId: string) => !currentBucketChoiceIds.includes(choiceId)
      );
    });
    return super
      .getChoicesThatChangedSinceLastSubmit(latestSubmitComponentState)
      .concat(removedChoices);
```